Shopify/shopify-cli
# [Scripts] Remove the need for `Script::Tasks::EnsureEnv`

### WHY are these changes introduced?

Fixes https://github.com/Shopify/script-service/issues/3659

### WHAT is this pull request doing?

Removes the usage of a `Task` and instead splits:
- the command line prompt, and
- the environment fresh-ness check

into their own sections.

### Current Questions
> The wants_to_run_against_shopify_org method

The only way to invoke this is to subclass the Task and then invoke it there. However, we could:
- make this public, we can just call this task to get this information.
- duplicate this code somewhere else to access it.

Currently, I've made this method public, but it doesn't feel great.

### How to test your changes?

- Create a new script `shopify script create`
- Push that script `shopify script push`
  - This should ask you the form questions since you have a fresh environment

- Attempt to push the script again `shopify script push`
  - This should NOT ask you the form questions since your env has already been populated.

### Update checklist
